src/app/shared/dot-save-on-deactivate-service/save-on-deactivate.ts

```diff
@@ -0,0 +1,10 @@
+import { Observable } from 'rxjs/Observable';
+import { DotDialog } from '../models/dot-confirmation/dot-confirmation.model';
+import { ResponseView } from 'dotcms-js/dotcms-js';
+
+export interface OnSaveDeactivate {
+    isModelChanged(): boolean;
```

Comment:
I think we should call this like:  ```shouldSaveBefore``` or something like that because in this particular case we know that we need to save if a model change but in other components it could be another reason.